modules/react-native-status/android/src/main/java/im/status/ethereum/module/RNSelectableTextInputViewManager.java

```diff
@@ -0,0 +1,117 @@
+        if(items != null) {
+            List<String> result = new ArrayList<String>(items.size());
+            for (int i = 0; i < items.size(); i++) {
+                result.add(items.getString(i));
+            }
+
+            this._menuItems = result.toArray(new String[items.size()]);
+
+            // PREVIOUS CODE
+            // registerSelectionListener(result.toArray(new String[items.size()]), reactViewGroup);
+        }
+
+    }
+
+    public void registerSelectionListener(final ReactEditText view) {
+        view.setCustomSelectionActionModeCallback(new Callback() {
+            @Override
+            public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+                // Called when action mode is first created. The menu supplied
+                // will be used to generate action buttons for the action mode
+                // Android Smart Linkify feature pushes extra options into the menu
+                // and would override the generated menu items
+                menu.clear();
+                for (int i = 0; i < _menuItems.length; i++) {
+                    menu.add(0, i, 0, _menuItems[i]);
+                }
+                return true;
+            }
+
+            @Override
+            public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+                return true;
+            }
+
+            @Override
+            public void onDestroyActionMode(ActionMode mode) {
+                // Called when an action mode is about to be exited and
+            }
+
+            @Override
+            public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+                int selectionStart = view.getSelectionStart();
+                int selectionEnd = view.getSelectionEnd();
+                String selectedText = view.getText().toString().substring(selectionStart, selectionEnd);
+
+                // Dispatch event
+                onSelectNativeEvent(view, item.getItemId(), selectedText, selectionStart, selectionEnd);
+
+                mode.finish();
+
+                return true;
+            }
+
+        });
+    }
+
+    public void onSelectNativeEvent(ReactEditText view, int eventType, String content, int selectionStart, int selectionEnd) {
+        WritableMap event = Arguments.createMap();
+        event.putInt("eventType", eventType);
+        event.putString("content", content);
+        event.putInt("selectionStart", selectionStart);
+        event.putInt("selectionEnd", selectionEnd);
+
+        // Dispatch
+        ReactContext reactContext = (ReactContext) view.getContext();
+        reactContext.getJSModule(RCTEventEmitter.class).receiveEvent(view.getId(), "topSelection", event);
+    }
+
+    @Override
+    public Map getExportedCustomDirectEventTypeConstants() {
+        return MapBuilder.builder()
+                         .put("topSelection", MapBuilder.of("registrationName","onSelection"))
+                         .build();
+    }
+}
```

Comment:
Can you please add a new line at the end of this file ?

Reply:
hi @ibrkhalil can u explain why we need a new line at the end of this file?

Reply:
Acts as a file terminator, And it makes our code base compatible with more tools
[Read](https://thoughtbot.com/blog/no-newline-at-end-of-file)